clients/cloud/java/src/main/java/io/confluent/examples/clients/ProducerExample.java

```diff
@@ -16,57 +16,86 @@
-import org.apache.kafka.clients.producer.Producer;
 import org.apache.kafka.clients.producer.ProducerRecord;
+import org.apache.kafka.clients.producer.ProducerConfig;
+import org.apache.kafka.clients.producer.Producer;
+import org.apache.kafka.connect.json.JsonSerializer;
+
+import org.apache.kafka.clients.admin.AdminClient;
+import org.apache.kafka.clients.admin.AdminClientConfig;
+import org.apache.kafka.clients.admin.NewTopic;
+
+import io.confluent.examples.clients.model.RecordJSON;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
 
 import java.io.FileInputStream;
 import java.io.IOException;
+import java.util.concurrent.ExecutionException;
+import org.apache.kafka.common.errors.TopicExistsException;
 import java.io.InputStream;
 import java.nio.file.Files;
 import java.nio.file.Paths;
-import java.util.Date;
 import java.util.Properties;
-import java.util.Random;
+import java.util.Collections;
+import java.util.Map;
 
 public class ProducerExample {
+
+  private static final ObjectMapper MAPPER = new ObjectMapper();
+
+  // Create topic in Confluent Cloud
+  public static void createTopic(final String topic,
+                          final int partitions,
+                          final int replication,
+                          final Properties cloudConfig) {
+      final NewTopic newTopic = new NewTopic(topic, partitions, (short) replication);
+      try (final AdminClient adminClient = AdminClient.create(cloudConfig)) {
+          adminClient.createTopics(Collections.singletonList(newTopic)).all().get();
+      } catch (final InterruptedException | ExecutionException e) {
+          // Ignore if TopicExistsException, which may be valid if topic exists
+          if (!(e.getCause() instanceof TopicExistsException)) {
+              throw new RuntimeException(e);
+          }
+      }
+  }
+
   public static void main(String[] args) throws IOException {
-    if (args.length != 3) {
-      System.out.println("Please provide command line arguments: configPath topic numEvents");
+    if (args.length != 2) {
+      System.out.println("Please provide command line arguments: configPath topic");
       System.exit(-1);
     }
 
-    long events = Long.parseLong(args[2]);
-    String topic = args[1];
-
-    // Load properties from disk.
+    // Load properties from file
     Properties props = loadConfig(args[0]);
 
-    // Add additional properties.
-    props.put("acks", "all");
-    props.put("retries", 200);
-    props.put("batch.size", 16384);
-    props.put("linger.ms", 100);
-    props.put("compression.type", "lz4");
-    props.put("buffer.memory", 33554432);
-    props.put("key.serializer", "org.apache.kafka.common.serialization.StringSerializer");
-    props.put("value.serializer", "org.apache.kafka.common.serialization.StringSerializer");
-
+    // Create topic if needed
+    String topic = args[1];
+    createTopic(topic, 12, 3, props);
```

Comment:
12 partitions? the other examples used one partition

Reply:
Ack